src/Conversion/ONNXToTOSA/ConvertONNXToTOSA.cpp

```diff
@@ -79,24 +51,37 @@ struct FrontendToTosaLoweringPass
 };
 
 void FrontendToTosaLoweringPass::runOnOperation() {
+  ModuleOp module = getOperation();
+  // Define final conversion target
   MLIRContext *context = &getContext();
   RewritePatternSet patterns(context);
   ConversionTarget target(*context);
 
+  // We use the type converter to legalize types before any conversion patterns
+  // are executed. This ensures that we do not need to trigger separate
+  // conversion failures.
   TypeConverter typeConverter;
-  typeConverter.addConversion([](Type type) { return type; });
-
+  typeConverter.addConversion([](Type type) -> Optional<Type> {
+    if (isa_tosa_signed_int(type) || isa_tosa_float(type))
+      return type;
+    return llvm::None;
+  });
+  typeConverter.addConversion([](TensorType type) -> Optional<Type> {
+    if (isa_tosa_signed_int(type.getElementType()) ||
```

Comment:
Why not `[&]` capture and then `typeConverter.isLegal(type.getElementType())`?

Reply:
ah nice!